### ISSUE TITLE
Name lookup 500 when subdomain isn't registered

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#531]
Seeing `500` error on name lookup  when subdomain isn't registered. 

http://localhost:6270/v1/names/foo.bar.id -> this works properly
http://localhost:6270/v1/names/cat.bar.id -> 500. 

I'm using `blockstack-run-scenario.blockstack_integration_tests.scenarios.subdomain_registrar` and  the `blockstack-venv` artifact from https://circleci.com/gh/blockstack/blockstack-core/372#artifacts/containers/0

I'd expect a `404` when a subdomain doesn't exist.

```
[2017-08-13 19:51:52,221] [DEBUG] [config:869] (16330.123145417678848) Add new field subdomain-resolution.subdomains_db
[2017-08-13 19:51:52,227] [DEBUG] [registrar:951] (16330.123145417678848) Registrar sleeping for 1
[2017-08-13 19:51:52,461] [DEBUG] [rpc:3697] (16330.123145421885440) 
full path: /v1/names/cat.id
method: GET
path: /v1/names/cat.id
qs: {}
headers:
origin: http://localhost:3000
accept-language: en-US,en;q=0.8
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36
host: localhost:6270
referer: http://localhost:3000/profiles/i/add-username/1EYfvtFJfRruBTQv2ERaHumLkDNN9qejxs/search

[2017-08-13 19:51:52,461] [DEBUG] [rpc:368] (16330.123145421885440) No authorization header
[2017-08-13 19:51:52,462] [DEBUG] [rpc:3736] (16330.123145421885440) No authentication needed
[2017-08-13 19:51:52,464] [DEBUG] [config:891] (16330.123145421885440) Change blockstack-client.client_version to 0.14.4.0
[2017-08-13 19:51:52,464] [DEBUG] [config:869] (16330.123145421885440) Add new field subdomain-resolution.subdomains_db
[2017-08-13 19:51:52,471] [DEBUG] [config:891] (16330.123145421885440) Change blockstack-client.client_version to 0.14.4.0
[2017-08-13 19:51:52,472] [DEBUG] [config:869] (16330.123145421885440) Add new field subdomain-resolution.subdomains_db
[2017-08-13 19:51:52,477] [DEBUG] [rpc:4744] (16330.123145421885440) Connect to API at localhost:6270
[2017-08-13 19:51:52,478] [DEBUG] [registrar:1014] (16330.123145421885440) Get queue state from /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.subdomain_registrar/client/queues.db
[2017-08-13 19:51:52,490] [DEBUG] [rpc:3697] (16330.123145426092032) 
full path: /v1/names/cat.foo.id
method: GET
path: /v1/names/cat.foo.id
qs: {}
headers:
origin: http://localhost:3000
accept-language: en-US,en;q=0.8
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36
host: localhost:6270
referer: http://localhost:3000/profiles/i/add-username/1EYfvtFJfRruBTQv2ERaHumLkDNN9qejxs/search

[2017-08-13 19:51:52,490] [DEBUG] [rpc:368] (16330.123145426092032) No authorization header
[2017-08-13 19:51:52,491] [DEBUG] [rpc:3736] (16330.123145426092032) No authentication needed
[2017-08-13 19:51:52,493] [DEBUG] [config:891] (16330.123145426092032) Change blockstack-client.client_version to 0.14.4.0
[2017-08-13 19:51:52,493] [DEBUG] [config:869] (16330.123145426092032) Add new field subdomain-resolution.subdomains_db
[2017-08-13 19:51:52,502] [DEBUG] [config:891] (16330.123145426092032) Change blockstack-client.client_version to 0.14.4.0
[2017-08-13 19:51:52,503] [DEBUG] [config:869] (16330.123145426092032) Add new field subdomain-resolution.subdomains_db
[2017-08-13 19:51:52,515] [DEBUG] [rpc:4744] (16330.123145426092032) Connect to API at localhost:6270
[2017-08-13 19:51:52,520] [DEBUG] [registrar:1014] (16330.123145426092032) Get queue state from /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.subdomain_registrar/client/queues.db
[2017-08-13 19:51:52,517] [DEBUG] [rpc:792] (16330.123145421885440) let's go!
[2017-08-13 19:51:52,531] [DEBUG] [rpc:792] (16330.123145426092032) let's go!
[2017-08-13 19:51:52,531] [DEBUG] [rpc:796] (16330.123145426092032) here too!
[2017-08-13 19:51:52,533] [DEBUG] [config:891] (16330.123145426092032) Change blockstack-client.client_version to 0.14.4.0
[2017-08-13 19:51:52,534] [DEBUG] [config:869] (16330.123145426092032) Add new field subdomain-resolution.subdomains_db
127.0.0.1 - - [13/Aug/2017 19:51:52] "GET /v1/names/cat.id HTTP/1.1" 404 -
[2017-08-13 19:51:52,576] [DEBUG] [subdomains:208] (16330.123145426092032) Fetching zonefiles in range (308, 379)
[2017-08-13 19:51:52,588] [ERROR] [subdomains:459] (16330.123145426092032) cat.foo.id
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/subdomains.py", line 457, in get_subdomain_info
    subdomain_obj = subdomain_db["{}.{}".format(subdomain, domain_fqa)]
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/subdomains.py", line 285, in __getitem__
    return self.get_subdomain_entry(fqn)
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/subdomains.py", line 166, in get_subdomain_entry
    raise SubdomainNotFound(fqn)
SubdomainNotFound: cat.foo.id
[2017-08-13 19:51:52,589] [ERROR] [subdomains:460] (16330.123145426092032) Raising SubdomainNotFound(cat) from exception cat.foo.id
[2017-08-13 19:51:52,589] [ERROR] [rpc:3776] (16330.123145426092032) cat
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/rpc.py", line 3773, in _dispatch
    return route_method( session, path_info, *route_args )
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/rpc.py", line 797, in GET_name_info
    subdomain_obj = subdomains.get_subdomain_info(subdomain, domain)
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.4.0-py2.7.egg/blockstack_client/subdomains.py", line 461, in get_subdomain_info
    raise SubdomainNotFound(subdomain)
SubdomainNotFound: cat
127.0.0.1 - - [13/Aug/2017 19:51:52] "GET /v1/names/cat.foo.id HTTP/1.1" 500 -
```

**Comment by kantai** [comment]
Okay, merged a fix in PR #532

**Comment by larrysalibra** [comment]
This is fixed for me. Thank you @kantai for the speedy turnaround!
